tests/link/numba/test_slinalg.py

```diff
@@ -102,3 +107,54 @@ def test_solve_triangular_raises_on_nan_inf(value):
         ValueError, match=re.escape("Non-numeric values (nan or inf) returned ")
     ):
         f(A_tri, b)
+
+
+@pytest.mark.parametrize(
+    "x, lower, exc",
+    [
+        (
+            set_test_value(
+                pt.dmatrix(),
+                (lambda x: x.T.dot(x))(rng.random(size=(3, 3)).astype("float64")),
+            ),
+            True,
+            None,
+        ),
+        (
+            set_test_value(
+                pt.lmatrix(),
+                (lambda x: x.T.dot(x))(
+                    rng.integers(1, 10, size=(3, 3)).astype("int64")
+                ),
+            ),
+            True,
+            None,
+        ),
+        (
+            set_test_value(
+                pt.dmatrix(),
+                (lambda x: x.T.dot(x))(rng.random(size=(3, 3)).astype("float64")),
+            ),
+            False,
+            UserWarning,
+        ),
+    ],
+)
```

Comment:
Since you've done quite a lot to get LAPACK's `xpotrf`multiple dtypes, don't you want to test on complex and float32 also?

Reply:
Yes, and complex types too. I just lazily copied the old test without updating it. Thanks for flagging this, I'll refactor the test.

Reply:
I refactored the tests so `float32` is covered. Complex is going to be a bit more complex (heh), because the ctype pointer hack I'm using to wrap the LAPACK calls doesn't work for complex inputs anymore. I remember hitting this problem in my own work a few months back, but I blocked it out of my mind. 

It's an edge case of an edge case so I'm not too worried about it, but I would prefer 100% coverage. It might mean actually figuring out how scipy binds the LAPACK functions to C code...

Reply:
Feel free to raise NotImplementedError for now